{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# ENIGMA GCTA"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Data is stored at bic.pasteur.fr. The pre-processing script is preproc.sh.\n",
    "\n",
    "<table>\n",
    "\n",
    "<tr>\n",
    "<th>Group</th>\n",
    "<th>N_total</th>\n",
    "<th>N_males</th>\n",
    "<th>N_females</th>\n",
    "<th>N_unspecified</th>\n",
    "<th>SNPs_total</th>\n",
    "<th>SNPs_missing</th>\n",
    "<th>Assembly</th>\n",
    "</tr>\n",
    "\n",
    "<tr>\n",
    "<td>Trend</td>\n",
    "<td>858</td>\n",
    "<td>381</td>\n",
    "<td>477</td>\n",
    "<td></td>\n",
    "<td>2450000</td>\n",
    "<td>471893</td>\n",
    "<td>hg18</td>\n",
    "</tr>\n",
    "\n",
    "<tr>\n",
    "<td>Ship</td>\n",
    "<td>963</td>\n",
    "<td>458</td>\n",
    "<td>505</td>\n",
    "<td></td>\n",
    "<td>900475</td>\n",
    "<td>3604</td>\n",
    "<td>hg18</td>\n",
    "</tr>\n",
    "\n",
    "<tr>\n",
    "<td>ADNI1</td>\n",
    "<td>757</td>\n",
    "<td>449</td>\n",
    "<td>308</td>\n",
    "<td></td>\n",
    "<td>620901</td>\n",
    "<td>7280</td>\n",
    "<td>hg18</td>\n",
    "</tr>\n",
    "\n",
    "<tr>\n",
    "<td>ADNI2</td>\n",
    "<td>432</td>\n",
    "<td>236</td>\n",
    "<td>196</td>\n",
    "<td></td>\n",
    "<td>730525</td>\n",
    "<td>43219</td>\n",
    "<td><b>hg19</b></td>\n",
    "</tr>\n",
    "\n",
    "<tr>\n",
    "<td>IMAGEN1</td>\n",
    "<td>705</td>\n",
    "<td>302</td>\n",
    "<td>318</td>\n",
    "<td>85</td>\n",
    "<td>582892</td>\n",
    "<td>5883</td>\n",
    "<td>hg18</td>\n",
    "</tr>\n",
    "\n",
    "<tr>\n",
    "<td>IMAGEN2</td>\n",
    "<td>993</td>\n",
    "<td>413</td>\n",
    "<td>436</td>\n",
    "<td>144</td>\n",
    "<td>557124</td>\n",
    "<td>26919</td>\n",
    "<td>hg18</td>\n",
    "</tr>\n",
    "\n",
    "<tr>\n",
    "<td>IMAGEN3</td>\n",
    "<td>393</td>\n",
    "<td>170</td>\n",
    "<td>150</td>\n",
    "<td>73</td>\n",
    "<td>557124</td>\n",
    "<td>1318</td>\n",
    "<td>hg18</td>\n",
    "</tr>\n",
    "\n",
    "<tr>\n",
    "<td>Osaka</td>\n",
    "<td>324</td>\n",
    "<td>158</td>\n",
    "<td>166</td>\n",
    "<td></td>\n",
    "<td>906295</td>\n",
    "<td>222</td>\n",
    "<td><b>hg19</b></td>\n",
    "</tr>\n",
    "\n",
    "<tr>\n",
    "<td>Lothian</td>\n",
    "<td>1005</td>\n",
    "<td>509</td>\n",
    "<td>496</td>\n",
    "<td></td>\n",
    "<td>542050</td>\n",
    "<td>1407</td>\n",
    "<td>hg18</td>\n",
    "</tr>\n",
    "\n",
    "<tr>\n",
    "<td><b>Total</b></td>\n",
    "<td>6430</td>\n",
    "<td></td>\n",
    "<td></td>\n",
    "<td></td>\n",
    "<td></td>\n",
    "<td></td>\n",
    "<td></td>\n",
    "</tr>\n",
    "\n",
    "</table>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#Statistical power for GCTA\n",
    "Using http://cnsgenomics.com/shiny/gctaPower/ we estimate that with a sample size of N=6430, we have > 81.24% power to detect $h^2$ > 0.14 (s.e., 0.0492)."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#Checking genome assembly\n",
    "\n",
    "Looking at http://may2009.archive.ensembl.org/Homo_sapiens/Variation/Summary?source=dbSNP;v=rs3094315\n",
    "\n",
    "ENIGMA2-GCTA-SHIP-TREND.bim\n",
    "bim:\n",
    "1\trs3094315\t0\t742429\tG\tA\n",
    "ncbi36:\n",
    "1:742429 (forward strand) \tJump to region in detail\n",
    "\n",
    "ENIGMA2-GCTA-SHIP.bim\n",
    "bim:\n",
    "1\trs10458597\t0\t554484\tT\tC\n",
    "ncbi36:\n",
    "1:554484 (forward strand) Jump to region in detail\n",
    "\n",
    "ADNI_cluster_01_forward_757LONI.bim\n",
    "bim:\n",
    "1\trs2691310\t0\t46844\t0\t0\n",
    "ncbi36:\n",
    "rs2691310: 1:46844 (forward strand) Jump to region in detail\n",
    "\n",
    "IMAGEN: all_subjects_all_snps_wave1.bim\n",
    "bim:\n",
    "1\trs4654470\t0\t4896086\tG\tA\n",
    "ncbi36:\n",
    "1:4896086 (forward strand) Jump to region in detail\n",
    "\n",
    "IMAGEN: all_subjects_all_snps_wave2.bim\n",
    "bim:\n",
    "1\trs945322\t0\t5606888\tC\tA\n",
    "ncbi36:\n",
    "1:5606888 (forward strand) Jump to region in detail\n",
    "\n",
    "##Get the genome position of a SNP in hg19:\n",
    "mysql --user=genome --host=genome-mysql.cse.ucsc.edu -A -D hg19 -e 'SELECT name,chromStart,chromEnd,alleles,alleleFreqs FROM snp138Common WHERE name = \"rs945322\"'\n",
    "\n",
    "##Get the genome position of a SNP in hg18:\n",
    "mysql --user=genome --host=genome-mysql.cse.ucsc.edu -A -D hg18 -e 'SELECT name,chrom,chromEnd FROM snp130 WHERE name = \"rs945322\"'\n",
    "\n",
    "All studies appear to be in hg18 with the exception of ADNI2 (ADNI_GO) and Osaka which are hg19"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "````\n",
    "2015_09SHIP/_Data_Germany-SHIP/ENIGMA2-GCTA-SHIP-SHIP_TREND/ENIGMA2-GCTA-SHIP.bim\n",
    "bim: rs10458597 1 554484 hg18: rs10458597 chr1 554484 hg19: rs10458597 564621 C,T, 0.976965,0.023035,\n",
    "bim: rs9629043 1 554636 hg18: rs9629043 chr1 554636 rs9629043 chrM 4224 hg19:\n",
    "bim: rs11510103 1 557616 hg18: rs11510103 chr1 557616 hg19:\n",
    "bim: rs12565286 1 711153 hg18: rs12565286 chr1 711153 hg19: rs12565286 721290 C,G, 0.040827,0.959173,\n",
    "bim: rs12082473 1 730720 hg18: rs12082473 chr1 730720 hg19: rs12082473 740857 A,G, 0.069470,0.930530,\n",
    "bim: rs3094315 1 742429 hg18: rs3094315 chr1 742429 hg19: rs3094315 752566 C,T, 0.265491,0.734509,\n",
    "bim: rs2286139 1 751595 hg18: rs2286139 chr1 751595 hg19: rs2286139 761732 A,G, 0.568086,0.431914,\n",
    "bim: rs11240776 1 755132 hg18: rs11240776 chr1 755132 hg19: rs11240776 765269 A,G, 0.988059,0.011941,\n",
    "bim: rs2980319 1 766985 hg18: rs2980319 chr1 766985 hg19: rs2980319 777122 A,T, 0.735724,0.264276,\n",
    "bim: rs2980300 1 775852 hg18: rs2980300 chr1 775852 hg19: rs2980300 785989 C,T, 0.591846,0.408154,\n",
    "\n",
    "2015_09SHIP/_Data_Germany-SHIP/ENIGMA2-GCTA-SHIP-SHIP_TREND/ENIGMA2-GCTA-SHIP-TREND.bim\n",
    "bim: rs4477212 1 72017 hg18: rs4477212 chr1 72017 hg19:\n",
    "bim: rs4000335 1 679049 hg18: rs4000335 chr1 143360 rs4000335 chr1 679049 rs4000335 chr11 138566 rs4000335 chr16 88742891 rs4000335 chr17_random 723152 rs4000335 chr19 169871 rs4000335 chr3 199412548 rs4000335 chr5 180666116 rs4000335 chr7 82571 hg19:\n",
    "bim: rs3094315 1 742429 hg18: rs3094315 chr1 742429 hg19: rs3094315 752566 C,T, 0.265491,0.734509,\n",
    "bim: rs3131972 1 742584 hg18: rs3131972 chr1 742584 hg19: rs3131972 752721 C,T, 0.648370,0.351630,\n",
    "bim: rs12124819 1 766409 hg18: rs12124819 chr1 766409 hg19: rs12124819 776546 A,G, 0.908778,0.091222,\n",
    "bim: rs11240777 1 788822 hg18: rs11240777 chr1 788822 hg19: rs11240777 798959 A,G,N, 0.443041,0.555831,0.001128,\n",
    "bim: rs6681049 1 789870 hg18: rs6681049 chr1 789870 hg19: rs6681049 800007 C,T, 0.913210,0.086790,\n",
    "bim: rs4970383 1 828418 hg18: rs4970383 chr1 828418 hg19: rs4970383 838555 G,T, 0.682505,0.317495,\n",
    "bim: rs4475691 1 836671 hg18: rs4475691 chr1 836671 hg19: rs4475691 846808 C,T, 0.755753,0.244247,\n",
    "bim: rs7537756 1 844113 hg18: rs7537756 chr1 844113 hg19: rs7537756 854250 A,G, 0.799520,0.200480,\n",
    "\n",
    "2016_01Imagen/all_subjects_all_snps_wave3/all_subjects_all_snps_wave3.bim\n",
    "bim: rs3094315 1 742429 hg18: rs3094315 chr1 742429 hg19: rs3094315 752566 C,T, 0.265491,0.734509,\n",
    "bim: rs12562034 1 758311 hg18: rs12562034 chr1 758311 hg19: rs12562034 768448 A,G, 0.187440,0.812560,\n",
    "bim: rs3934834 1 995669 hg18: rs3934834 chr1 995669 hg19: rs3934834 1005806 C,T, 0.786881,0.213119,\n",
    "bim: rs9442372 1 1008567 hg18: rs9442372 chr1 1008567 hg19: rs9442372 1018704 A,G, 0.396066,0.603934,\n",
    "bim: rs3737728 1 1011278 hg18: rs3737728 chr1 1011278 hg19: rs3737728 1021415 C,T, 0.780997,0.219003,\n",
    "bim: rs11260588 1 1011521 hg18: rs11260588 chr1 1011521 hg19: rs11260588 1021658 A,G, 0.070132,0.929868,\n",
    "bim: rs6687776 1 1020428 hg18: rs6687776 chr1 1020428 hg19: rs6687776 1030565 C,T, 0.705162,0.294838,\n",
    "bim: rs9651273 1 1021403 hg18: rs9651273 chr1 1021403 hg19: rs9651273 1031540 A,G, 0.126440,0.873560,\n",
    "bim: rs4970405 1 1038818 hg18: rs4970405 chr1 1038818 hg19: rs4970405 1048955 A,G, 0.887597,0.112403,\n",
    "bim: rs12726255 1 1039813 hg18: rs12726255 chr1 1039813 hg19: rs12726255 1049950 A,G, 0.709741,0.290259,\n",
    "\n",
    "2016_01Imagen/all_subjects_all_snps_wave2/all_subjects_all_snps_wave2.bim\n",
    "bim: rs3094315 1 742429 hg18: rs3094315 chr1 742429 hg19: rs3094315 752566 C,T, 0.265491,0.734509,\n",
    "bim: rs12562034 1 758311 hg18: rs12562034 chr1 758311 hg19: rs12562034 768448 A,G, 0.187440,0.812560,\n",
    "bim: rs3934834 1 995669 hg18: rs3934834 chr1 995669 hg19: rs3934834 1005806 C,T, 0.786881,0.213119,\n",
    "bim: rs9442372 1 1008567 hg18: rs9442372 chr1 1008567 hg19: rs9442372 1018704 A,G, 0.396066,0.603934,\n",
    "bim: rs3737728 1 1011278 hg18: rs3737728 chr1 1011278 hg19: rs3737728 1021415 C,T, 0.780997,0.219003,\n",
    "bim: rs11260588 1 1011521 hg18: rs11260588 chr1 1011521 hg19: rs11260588 1021658 A,G, 0.070132,0.929868,\n",
    "bim: rs6687776 1 1020428 hg18: rs6687776 chr1 1020428 hg19: rs6687776 1030565 C,T, 0.705162,0.294838,\n",
    "bim: rs9651273 1 1021403 hg18: rs9651273 chr1 1021403 hg19: rs9651273 1031540 A,G, 0.126440,0.873560,\n",
    "bim: rs4970405 1 1038818 hg18: rs4970405 chr1 1038818 hg19: rs4970405 1048955 A,G, 0.887597,0.112403,\n",
    "bim: rs12726255 1 1039813 hg18: rs12726255 chr1 1039813 hg19: rs12726255 1049950 A,G, 0.709741,0.290259,\n",
    "\n",
    "2016_01Imagen/all_subjects_all_snps_wave1/all_subjects_all_snps_wave1.bim\n",
    "bim: rs12354060 1 10004 hg18: rs12354060 chr1 10004 rs12354060 chr15 100328546 rs12354060 chr19 12749 rs12354060 chr9 10254 hg19:\n",
    "bim: rs12184279 1 707348 hg18: rs12184279 chr1 707348 hg19: rs12184279 717485 A,C, 0.016943,0.983057,\n",
    "bim: rs12564807 1 724325 hg18: rs12564807 chr1 724325 hg19: rs12564807 734462 A,G, 0.760066,0.239934,\n",
    "bim: rs3094315 1 742429 hg18: rs3094315 chr1 742429 hg19: rs3094315 752566 C,T, 0.265491,0.734509,\n",
    "bim: rs3115860 1 743268 hg18: rs3115860 chr1 743268 rs3115860 chr8 317740 hg19:\n",
    "bim: rs3115850 1 751010 hg18: rs3115850 chr1 751010 hg19: rs3115850 761147 A,G, 0.285941,0.714059,\n",
    "bim: rs12562034 1 758311 hg18: rs12562034 chr1 758311 hg19: rs12562034 768448 A,G, 0.187440,0.812560,\n",
    "bim: rs7515489 1 758845 hg18: rs7515489 chr1 758845 hg19:\n",
    "bim: rs12124819 1 766409 hg18: rs12124819 chr1 766409 hg19: rs12124819 776546 A,G, 0.908778,0.091222,\n",
    "bim: rs12086311 1 798632 hg18: rs12086311 chr1 798632 hg19:\n",
    "\n",
    "2016_02Lothian/ENIGMA_GCTA_LBC1936/LBC36_clean_231009.bim\n",
    "bim: rs3094315 1 742429 hg18: rs3094315 chr1 742429 hg19: rs3094315 752566 C,T, 0.265491,0.734509,\n",
    "bim: rs12562034 1 758311 hg18: rs12562034 chr1 758311 hg19: rs12562034 768448 A,G, 0.187440,0.812560,\n",
    "bim: rs4475691 1 836671 hg18: rs4475691 chr1 836671 hg19: rs4475691 846808 C,T, 0.755753,0.244247,\n",
    "bim: rs28705211 1 890368 hg18: rs28705211 chr1 890368 hg19: rs28705211 900505 C,G, 0.198406,0.801594,\n",
    "bim: rs13303118 1 908247 hg18: rs13303118 chr1 908247 hg19: rs13303118 918384 G,T, 0.514918,0.485082,\n",
    "bim: rs9777703 1 918699 hg18: rs9777703 chr1 918699 hg19: rs9777703 928836 C,T, 0.129071,0.870929,\n",
    "bim: rs3121567 1 933331 hg18: rs3121567 chr1 933331 hg19: rs3121567 943468 A,G, 0.153420,0.846580,\n",
    "bim: rs3934834 1 995669 hg18: rs3934834 chr1 995669 hg19: rs3934834 1005806 C,T, 0.786881,0.213119,\n",
    "bim: rs9442372 1 1008567 hg18: rs9442372 chr1 1008567 hg19: rs9442372 1018704 A,G, 0.396066,0.603934,\n",
    "bim: rs3737728 1 1011278 hg18: rs3737728 chr1 1011278 hg19: rs3737728 1021415 C,T, 0.780997,0.219003,\n",
    "\n",
    "2016_01Osaka/Handai_GCTA/Handai_GCTA.bim\n",
    "bim: rs10458597 1 564621 hg18: rs10458597 chr1 554484 hg19: rs10458597 564621 C,T, 0.976965,0.023035,\n",
    "bim: rs12565286 1 721290 hg18: rs12565286 chr1 711153 hg19: rs12565286 721290 C,G, 0.040827,0.959173,\n",
    "bim: rs12082473 1 740857 hg18: rs12082473 chr1 730720 hg19: rs12082473 740857 A,G, 0.069470,0.930530,\n",
    "bim: rs3094315 1 752566 hg18: rs3094315 chr1 742429 hg19: rs3094315 752566 C,T, 0.265491,0.734509,\n",
    "bim: rs2286139 1 761732 hg18: rs2286139 chr1 751595 hg19: rs2286139 761732 A,G, 0.568086,0.431914,\n",
    "bim: rs11240776 1 765269 hg18: rs11240776 chr1 755132 hg19: rs11240776 765269 A,G, 0.988059,0.011941,\n",
    "bim: rs2980319 1 777122 hg18: rs2980319 chr1 766985 hg19: rs2980319 777122 A,T, 0.735724,0.264276,\n",
    "bim: rs2980300 1 785989 hg18: rs2980300 chr1 775852 hg19: rs2980300 785989 C,T, 0.591846,0.408154,\n",
    "bim: rs2905036 1 792480 hg18: rs2905036 chr1 782343 hg19: rs2905036 792480 A,G, 0.974588,0.025412,\n",
    "bim: rs11240777 1 798959 hg18: rs11240777 chr1 788822 hg19: rs11240777 798959 A,G,N, 0.443041,0.555831,0.001128,\n",
    "\n",
    "2015_10ADNI/ADNI_1_GWAS_Plink/ADNI_cluster_01_forward_757LONI.bim\n",
    "bim: rs12354060 1 10004 hg18: rs12354060 chr1 10004 rs12354060 chr15 100328546 rs12354060 chr19 12749 rs12354060 chr9 10254 hg19:\n",
    "bim: rs2691310 1 46844 hg18: rs2691310 chr1 46844 rs2691310 chr15 100291725 rs2691310 chr19 49556 hg19:\n",
    "bim: rs2531266 1 59415 hg18: rs2531266 chr1 59415 rs2531266 chr15 100280324 rs2531266 chr19 62140 hg19:\n",
    "bim: rs4124251 1 97215 hg18: rs4124251 chr1 97215 rs4124251 chr1 243530 rs4124251 chr16 88793495 rs4124251 chr4 119803080 hg19:\n",
    "bim: rs8179466 1 224176 hg18: rs8179466 chr1 224176 rs8179466 chr16 88818746 hg19:\n",
    "bim: rs6603779 1 227744 hg18: rs6603779 chr1 227744 rs6603779 chr2 242750854 hg19:\n",
    "bim: rs28446478 1 576058 hg18: rs28446478 chr1 394133 rs28446478 chr1 576058 rs28446478 chr5 180763424 rs28446478 chr6 170830396 rs28446478 chr8 68789 hg19:\n",
    "bim: rs12029736 1 696231 hg18: rs12029736 chr1 696231 rs12029736 chr11 155765 rs12029736 chr16 88725825 rs12029736 chr19 186948 rs12029736 chr3 199395314 rs12029736 chr5 180648873 rs12029736 chr7 99635 rs12029736 chr9 140257184 hg19:\n",
    "bim: rs12184279 1 707348 hg18: rs12184279 chr1 707348 hg19: rs12184279 717485 A,C, 0.016943,0.983057,\n",
    "bim: rs2977656 1 719811 hg18: rs2977656 chr1 719811 hg19: rs2977656 729948 A,G, 0.021327,0.978673,\n",
    "\n",
    "2015_10ADNI/ADNI_GO_2_OmniExpress/ADNI_GO_2_Forward_Bin.bim\n",
    "bim: rs4477212 1 82154 hg18: rs4477212 chr1 72017 hg19:\n",
    "bim: rs3094315 1 752566 hg18: rs3094315 chr1 742429 hg19: rs3094315 752566 C,T, 0.265491,0.734509,\n",
    "bim: rs3131972 1 752721 hg18: rs3131972 chr1 742584 hg19: rs3131972 752721 C,T, 0.648370,0.351630,\n",
    "bim: rs12562034 1 768448 hg18: rs12562034 chr1 758311 hg19: rs12562034 768448 A,G, 0.187440,0.812560,\n",
    "bim: rs12124819 1 776546 hg18: rs12124819 chr1 766409 hg19: rs12124819 776546 A,G, 0.908778,0.091222,\n",
    "bim: rs11240777 1 798959 hg18: rs11240777 chr1 788822 hg19: rs11240777 798959 A,G,N, 0.443041,0.555831,0.001128,\n",
    "bim: rs6681049 1 800007 hg18: rs6681049 chr1 789870 hg19: rs6681049 800007 C,T, 0.913210,0.086790,\n",
    "bim: rs4970383 1 838555 hg18: rs4970383 chr1 828418 hg19: rs4970383 838555 G,T, 0.682505,0.317495,\n",
    "bim: rs4475691 1 846808 hg18: rs4475691 chr1 836671 hg19: rs4475691 846808 C,T, 0.755753,0.244247,\n",
    "bim: rs7537756 1 854250 hg18: rs7537756 chr1 844113 hg19: rs7537756 854250 A,G, 0.799520,0.200480,\n",
    "````"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Recover SNP lists, positions, alleles, frequencies\n",
    "* http://hgdownload.cse.ucsc.edu/goldenPath/hg18/database/snp130Seq.txt.gz\n",
    "* http://hgdownload.cse.ucsc.edu/goldenPath/hg19/database/snp138Common.txt.gz\n",
    "* http://hgdownload.cse.ucsc.edu/goldenPath/hg19/database/snp144Common.txt.gz"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "# Using liftOver to move everything to hg19\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "liftOver will just move rs numbers from hg18 to hg19, for example, but it will not do all the rest... UCSC people provide two other scripts for that: LiftMap.py and LiftRsNumber.py.\n",
    "All the process and information is described here: http://genome.sph.umich.edu/wiki/LiftOver\n",
    "There are two adviced methods to lift a complete ped/map plink dataset. The 1s uses liftMap.py and will only lift positions but not names (it lifts, however, most of the genome). The 2nd one uses LiftRsNumber.py and lifts both positions and names.\n",
    "\n",
    "LiftMpa.py starts by converting map files to UCSD bed files. However, I only have bim files. bim files contain more or less the same information as map files. I then first converted bim to map.\n",
    "\n",
    "plink map files look like:\n",
    "\n",
    "    1  rs123456  0  1234555\n",
    "    1  rs234567  0  1237793\n",
    "\n",
    "plink bim files look like:\n",
    "\n",
    "    21  rs11511647  0   26765   A   T\n",
    "    X   rs3883674   0   32380   C   G\n",
    "\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "Merging all datasets, may problems appear:\n",
    "\n",
    "````\n",
    "1=\"SHIP\"     906295\n",
    "2=\"TREND\"   2389858\n",
    "3=\"ADNI1\"    600310\n",
    "4=\"ADNI2\"    730525\n",
    "5=\"IMAGEN1\"  567454\n",
    "6=\"IMAGEN2\"  543611\n",
    "7=\"IMAGEN3\"  543611\n",
    "8=\"OSAKA\"    906295\n",
    "9=\"LOTHIAN\"  529015\n",
    "\n",
    "1_2: diff. pos |      8 misMatching SNPs    SHIP-TREND\n",
    "1_3: diff. pos |  62100 mismatching SNPs    SHIP-ADNI1\n",
    "1_4: diff. pos | 118685 mismatching SNPs    SHIP-ADNI2\n",
    "1_5: diff. pos |     12 mismatching SNPs    SHIP-IMAGEN1\n",
    "1_6: diff. pos |      1 mismatching SNPs    SHIP-IMAGEN2\n",
    "1_7: diff. pos |      1 mismatching SNPs    SHIP-IMAGEN3\n",
    "1_8: diff. pos |     43 mismatching SNPs    SHIP-OSAKA\n",
    "1_9: diff. pos | 309633 mismatching SNPs    SHIP-LOTHIAN\n",
    "2_3: diff. pos |  30839 mismatching SNPs   TREND-ADNI1\n",
    "2_4: diff. pos |  36494 mismatching SNPs   TREND-ADNI2\n",
    "2_5:           |                           TREND-IMAGEN1\n",
    "2_6:           |                           TREND-IMAGEN2\n",
    "2_7:           |                           TREND-IMAGEN3\n",
    "2_8: diff. pos |    129 mismatching SNPs   TREND-OSAKA\n",
    "2_9: diff. pos | 152142 mismatching SNPs   TREND-LOTHIAN\n",
    "3_4: diff. pos |                           ADNI1-ADNI2\n",
    "3_5: diff. pos | 104933 mismatching SNPs   ADNI1-IMAGEN1\n",
    "3_6: diff. pos | 101943 mismatching SNPs   ADNI1-IMAGEN2\n",
    "3_7: diff. pos | 101921 mismatching SNPs   ADNI1-IMAGEN3\n",
    "3_8: diff. pos |  30838 mismatching SNPs   ADNI1-OSAKA\n",
    "3_9: diff. pos | 529015 mismatching SNPs   ADNI1-LOTHIAN\n",
    "4_5: diff. pos |  61734 mismatching SNPs   ADNI2-IMAGEN1\n",
    "4_6: diff. pos |  59516 mismatching SNPs   ADNI2-IMAGEN2\n",
    "4_7: diff. pos |  59486 mismatching SNPs   ADNI2-IMAGEN3\n",
    "4_8: diff. pos |  37382 mismatching SNPs   ADNI2-OSAKA\n",
    "4_9: diff. pos | 313546 mismatching SNPs   ADNI2-LOTHIAN\n",
    "5_6:           |                         IMAGEN1-IMAGEN2\n",
    "5_7:           |                         IMAGEN1-IMAGEN3\n",
    "5_8: diff. pos |     24 mismatching SNPs IMAGEN1-OSAKA\n",
    "5_9: diff. pos | 521031 mismatching SNPs IMAGEN1-LOTHIAN\n",
    "6_7:           |                         IMAGEN2-IMAGEN3\n",
    "6_8: diff. pos |     24 mismatching SNPs IMAGEN2-OSAKA\n",
    "6_9: diff. pos | 508108 mismatching SNPs IMAGEN2-LOTHIAN\n",
    "7_8: diff. pos |     24 mismatching SNPs IMAGEN3-OSAKA\n",
    "7_9: diff. pos | 508105 mismatching SNPs IMAGEN3-LOTHIAN\n",
    "8_9: diff. pos | 151975 mismatching SNPs   OSAKA-LOTHIAN\n",
    "````\n",
    "\n",
    "````\n",
    "---------------------\n",
    "| |1|2|3|4|5|6|7|8|9|\n",
    "|1|@|@| | |@|@|@|@| |\n",
    "|2|@|@| | |@|@|@|@| |\n",
    "|3| | |@|@| | | | | |\n",
    "|4| | |@|@| | | | | |\n",
    "|5|@|@| | |@|@|@|@| |\n",
    "|6|@|@| | |@|@|@|@| |\n",
    "|7|@|@| | |@|@|@|@| |\n",
    "|8|@|@| | |@|@|@|@| |\n",
    "|9| | | | | | | | |@|\n",
    "---------------------\n",
    "````\n",
    "\n",
    "LOTHIAN is encoded as 1234 instead of ACGT, that's why it has so many mistmatches. This is the script I used to compare the genotypes:\n",
    "\n",
    "````\n",
    "for i in \"1 2 A C\" \"1 2 C A\" \"1 2 G T\" \"1 2 T G\" \"1 3 A G\" \"1 3 C T\" \"1 3 G A\" \"1 3 T C\" \"2 1 A C\" \"2 1 C A\" \"2 1 G T\" \"2 1 T G\" \"2 4 A G\" \"2 4 C T\" \"2 4 G A\" \"2 4 T C\" \"3 1 A G\" \"3 1 C T\" \"3 1 G A\" \"3 1 T C\" \"3 4 A C\" \"3 4 C A\" \"3 4 G T\" \"3 4 T G\" \"4 2 A G\" \"4 2 C T\" \"4 2 G A\" \"4 2 T C\" \"4 3 A C\" \"4 3 C A\" \"4 3 G T\" \"4 3 T G\";do echo $(awk 'FNR==NR{arr[$2]=$5\" \"$6} FNR!=NR{if(arr[$2]) print arr[$2],$5\" \"$6}' LOTHIAN/LOTHIAN.updt.hg19.bim ADNI1/ADNI1.updt.hg19.bim|grep \"$i\"|wc -l) $i;done|sort -n\n",
    "0 1 2 G T\n",
    "0 1 2 T G\n",
    "0 1 3 C T\n",
    "0 1 3 T C\n",
    "0 2 1 G T\n",
    "0 2 1 T G\n",
    "0 2 4 A G\n",
    "0 2 4 G A\n",
    "0 3 1 C T\n",
    "0 3 1 T C\n",
    "0 3 4 A C\n",
    "0 3 4 C A\n",
    "0 4 2 A G\n",
    "0 4 2 G A\n",
    "0 4 3 A C\n",
    "0 4 3 C A\n",
    "482 1 2 C A\n",
    "487 4 3 G T\n",
    "512 2 1 A C\n",
    "517 3 4 T G\n",
    "2029 4 2 C T\n",
    "2115 1 3 G A\n",
    "2207 2 4 T C\n",
    "2286 3 1 A G\n",
    "23065 3 4 G T\n",
    "23205 2 1 C A\n",
    "24937 4 3 T G\n",
    "25197 1 2 A C\n",
    "97768 3 1 G A\n",
    "97872 2 4 C T\n",
    "111559 1 3 A G\n",
    "112479 4 2 T C\n",
    "````\n",
    "\n",
    "````\n",
    "for i in \"1 2 A C\" \"1 2 C A\" \"1 2 G T\" \"1 2 T G\" \"1 3 A G\" \"1 3 C T\" \"1 3 G A\" \"1 3 T C\" \"2 1 A C\" \"2 1 C A\" \"2 1 G T\" \"2 1 T G\" \"2 4 A G\" \"2 4 C T\" \"2 4 G A\" \"2 4 T C\" \"3 1 A G\" \"3 1 C T\" \"3 1 G A\" \"3 1 T C\" \"3 4 A C\" \"3 4 C A\" \"3 4 G T\" \"3 4 T G\" \"4 2 A G\" \"4 2 C T\" \"4 2 G A\" \"4 2 T C\" \"4 3 A C\" \"4 3 C A\" \"4 3 G T\" \"4 3 T G\";do echo $(awk 'FNR==NR{arr[$2]=$5\" \"$6} FNR!=NR{if(arr[$2]) print arr[$2],$5\" \"$6}' LOTHIAN/LOTHIAN.updt.hg19.bim IMAGEN2/IMAGEN2.updt.hg19.bim|grep \"$i\"|wc -l) $i;done|sort -n\n",
    "86 3 4 A C\n",
    "95 1 2 G T\n",
    "98 2 1 T G\n",
    "98 4 3 C A\n",
    "324 1 2 C A\n",
    "329 4 3 G T\n",
    "356 2 1 A C\n",
    "361 3 4 T G\n",
    "396 2 4 A G\n",
    "398 4 2 G A\n",
    "400 1 3 C T\n",
    "473 3 1 T C\n",
    "1384 4 2 C T\n",
    "1445 1 3 G A\n",
    "1485 2 4 T C\n",
    "1533 3 1 A G\n",
    "4212 3 4 C A\n",
    "4253 2 1 G T\n",
    "4593 4 3 A C\n",
    "4630 1 2 T G\n",
    "17919 2 4 G A\n",
    "18041 3 4 G T\n",
    "18123 2 1 C A\n",
    "18151 3 1 C T\n",
    "19433 4 3 T G\n",
    "19658 1 2 A C\n",
    "20684 1 3 T C\n",
    "20835 4 2 A G\n",
    "76433 3 1 G A\n",
    "76702 2 4 C T\n",
    "87319 1 3 A G\n",
    "87859 4 2 T C\n",
    "\n",
    "````"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "All genotypes converted to ACGT. New merge results\n",
    "\n",
    "````\n",
    "1_2: diff. pos |      8 misMatching SNPs     43   SHIP-TREND\n",
    "1_3: diff. pos |  62100 mismatching SNPs  30838   SHIP-ADNI1\n",
    "1_4: diff. pos | 118685 mismatching SNPs  37382   SHIP-ADNI2\n",
    "1_5: diff. pos |     12 mismatching SNPs     24   SHIP-IMAGEN1\n",
    "1_6: diff. pos |      1 mismatching SNPs     24   SHIP-IMAGEN2\n",
    "1_7: diff. pos |      1 mismatching SNPs     24   SHIP-IMAGEN3\n",
    "1_8: diff. pos |     43 mismatching SNPs      0   SHIP-OSAKA\n",
    "1_9: diff. pos | 309633 mismatching SNPs  29651   SHIP-LOTHIAN\n",
    "2_3: diff. pos |  30839 mismatching SNPs  62100  TREND-ADNI1\n",
    "2_4: diff. pos |  36494 mismatching SNPs 118685  TREND-ADNI2\n",
    "2_5:           |                             12  TREND-IMAGEN1\n",
    "2_6:           |                              1  TREND-IMAGEN2\n",
    "2_7:           |                              1  TREND-IMAGEN3\n",
    "2_8: diff. pos |    129 mismatching SNPs     43  TREND-OSAKA\n",
    "2_9: diff. pos | 152142 mismatching SNPs  59460  TREND-LOTHIAN\n",
    "3_4: diff. pos |                              0  ADNI1-ADNI2\n",
    "3_5: diff. pos | 104933 mismatching SNPs 104933  ADNI1-IMAGEN1\n",
    "3_6: diff. pos | 101943 mismatching SNPs 101943  ADNI1-IMAGEN2\n",
    "3_7: diff. pos | 101921 mismatching SNPs 101921  ADNI1-IMAGEN3\n",
    "3_8: diff. pos |  30838 mismatching SNPs  30838  ADNI1-OSAKA\n",
    "3_9: diff. pos | 529015 mismatching SNPs      0  ADNI1-LOTHIAN\n",
    "4_5: diff. pos |  61734 mismatching SNPs  61734  ADNI2-IMAGEN1\n",
    "4_6: diff. pos |  59516 mismatching SNPs  59516  ADNI2-IMAGEN2\n",
    "4_7: diff. pos |  59486 mismatching SNPs  59486  ADNI2-IMAGEN3\n",
    "4_8: diff. pos |  37382 mismatching SNPs  37382  ADNI2-OSAKA\n",
    "4_9: diff. pos | 313546 mismatching SNPs      0  ADNI2-LOTHIAN\n",
    "5_6:           |                              0 IMAGEN1-IMAGEN2\n",
    "5_7:           |                              0 IMAGEN1-IMAGEN3\n",
    "5_8: diff. pos |     24 mismatching SNPs     24 IMAGEN1-OSAKA\n",
    "5_9: diff. pos | 521031 mismatching SNPs  99205 IMAGEN1-LOTHIAN\n",
    "6_7:           |                              0 IMAGEN2-IMAGEN3\n",
    "6_8: diff. pos |     24 mismatching SNPs     24 IMAGEN2-OSAKA\n",
    "6_9: diff. pos | 508108 mismatching SNPs  97322 IMAGEN2-LOTHIAN\n",
    "7_8: diff. pos |     24 mismatching SNPs     24 IMAGEN3-OSAKA\n",
    "7_9: diff. pos | 508105 mismatching SNPs  97322 IMAGEN3-LOTHIAN\n",
    "8_9: diff. pos | 151975 mismatching SNPs  29651 OSAKA-LOTHIAN\n",
    "\n",
    "---------------------\n",
    "| |1|2|3|4|5|6|7|8|9|\n",
    "|1|@|@|.|.|@|@|@|@|.|\n",
    "|2|@|@|-|-|@|@|@|@|-|\n",
    "|3|.|-|@|@| | | |.|@|\n",
    "|4|.|-|@|@| | | | |@|\n",
    "|5|@|@| | |@|@|@|@| |\n",
    "|6|@|@| | |@|@|@|@| |\n",
    "|7|@|@| | |@|@|@|@| |\n",
    "|8|@|@|.| |@|@|@|@|-|\n",
    "|9|.|-|@|@| | | |-|@|\n",
    "---------------------\n",
    "\n",
    "---------------------\n",
    "| |1|2|5|6|7|8|9|3|4|\n",
    "|1|@|@|@|@|@|@|.|.|.|\n",
    "|2|@|@|@|@|@|@|-|-|-|\n",
    "|5|@|@|@|@|@|@| | | |\n",
    "|6|@|@|@|@|@|@| | | |\n",
    "|7|@|@|@|@|@|@| | | |\n",
    "|8|@|@|@|@|@|@|-|.| |\n",
    "|9|.|-| | | |-|@|@|@|\n",
    "|3|.|-| | | |.|@|@|@|\n",
    "|4|.|-| | | | |@|@|@|\n",
    "---------------------\n",
    "1,2,5,6,7,8: SHIP, TREND, IMAGEN1, IMAGEN2, IMAGEN3, OSAKA\n",
    "3,4,9: ADNI1, ADNI2, LOTHIAN\n",
    "````"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "collapsed": true
   },
   "source": [
    "Accounting for problem SNPS when merging multiple plink files:\n",
    "https://www.biostars.org/p/101191/\n",
    "\n",
    "awk variables:\n",
    "http://www.thegeekstuff.com/2010/01/8-powerful-awk-built-in-variables-fs-ofs-rs-ors-nr-nf-filename-fnr"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.11"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
