{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#ENIGMA GCTA"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Data is stored at bic.pasteur.fr. The pre-processing script is preproc.sh.\n",
    "\n",
    "<table>\n",
    "\n",
    "<tr>\n",
    "<th>Group</th>\n",
    "<th>N_total</th>\n",
    "<th>N_males</th>\n",
    "<th>N_females</th>\n",
    "<th>N_unspecified</th>\n",
    "<th>SNPs_total</th>\n",
    "<th>SNPs_missing</th>\n",
    "<th>Assembly</th>\n",
    "</tr>\n",
    "\n",
    "<tr>\n",
    "<td>Ship</td>\n",
    "<td>858</td>\n",
    "<td>381</td>\n",
    "<td>477</td>\n",
    "<td></td>\n",
    "<td>2450000</td>\n",
    "<td>471893</td>\n",
    "<td>hg18</td>\n",
    "</tr>\n",
    "\n",
    "<tr>\n",
    "<td>Trend</td>\n",
    "<td>963</td>\n",
    "<td>458</td>\n",
    "<td>505</td>\n",
    "<td></td>\n",
    "<td>900475</td>\n",
    "<td>3604</td>\n",
    "<td>hg18</td>\n",
    "</tr>\n",
    "\n",
    "<tr>\n",
    "<td>ADNI1</td>\n",
    "<td>757</td>\n",
    "<td>449</td>\n",
    "<td>308</td>\n",
    "<td></td>\n",
    "<td>620901</td>\n",
    "<td>7280</td>\n",
    "<td>hg18</td>\n",
    "</tr>\n",
    "\n",
    "<tr>\n",
    "<td>ADNI2</td>\n",
    "<td>432</td>\n",
    "<td>236</td>\n",
    "<td>196</td>\n",
    "<td></td>\n",
    "<td>730525</td>\n",
    "<td>43219</td>\n",
    "<td><b>hg19</b></td>\n",
    "</tr>\n",
    "\n",
    "<tr>\n",
    "<td>IMAGEN1</td>\n",
    "<td>705</td>\n",
    "<td>302</td>\n",
    "<td>318</td>\n",
    "<td>85</td>\n",
    "<td>582892</td>\n",
    "<td>5883</td>\n",
    "<td>hg18</td>\n",
    "</tr>\n",
    "\n",
    "<tr>\n",
    "<td>IMAGEN2</td>\n",
    "<td>993</td>\n",
    "<td>413</td>\n",
    "<td>436</td>\n",
    "<td>144</td>\n",
    "<td>557124</td>\n",
    "<td>26919</td>\n",
    "<td>hg18</td>\n",
    "</tr>\n",
    "\n",
    "<tr>\n",
    "<td>IMAGEN3</td>\n",
    "<td>393</td>\n",
    "<td>170</td>\n",
    "<td>150</td>\n",
    "<td>73</td>\n",
    "<td>557124</td>\n",
    "<td>1318</td>\n",
    "<td>hg18</td>\n",
    "</tr>\n",
    "\n",
    "<tr>\n",
    "<td>Osaka</td>\n",
    "<td>324</td>\n",
    "<td>158</td>\n",
    "<td>166</td>\n",
    "<td></td>\n",
    "<td>906295</td>\n",
    "<td>222</td>\n",
    "<td><b>hg19</b></td>\n",
    "</tr>\n",
    "\n",
    "<tr>\n",
    "<td>Lothian</td>\n",
    "<td>1005</td>\n",
    "<td>509</td>\n",
    "<td>496</td>\n",
    "<td></td>\n",
    "<td>542050</td>\n",
    "<td>1407</td>\n",
    "<td>hg18</td>\n",
    "</tr>\n",
    "\n",
    "<tr>\n",
    "<td><b>Total</b></td>\n",
    "<td>6430</td>\n",
    "<td></td>\n",
    "<td></td>\n",
    "<td></td>\n",
    "<td></td>\n",
    "<td></td>\n",
    "<td></td>\n",
    "</tr>\n",
    "\n",
    "</table>"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#Statistical power for GCTA\n",
    "Using http://cnsgenomics.com/shiny/gctaPower/ we estimate that with a sample size of N=6430, we have > 81.24% power to detect $h^2$ > 0.14 (s.e., 0.0492)."
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "#Checking genome assembly\n",
    "\n",
    "Looking at http://may2009.archive.ensembl.org/Homo_sapiens/Variation/Summary?source=dbSNP;v=rs3094315\n",
    "\n",
    "ENIGMA2-GCTA-SHIP-TREND.bim\n",
    "bim:\n",
    "1\trs3094315\t0\t742429\tG\tA\n",
    "ncbi36:\n",
    "1:742429 (forward strand) \tJump to region in detail\n",
    "\n",
    "ENIGMA2-GCTA-SHIP.bim\n",
    "bim:\n",
    "1\trs10458597\t0\t554484\tT\tC\n",
    "ncbi36:\n",
    "1:554484 (forward strand) Jump to region in detail\n",
    "\n",
    "ADNI_cluster_01_forward_757LONI.bim\n",
    "bim:\n",
    "1\trs2691310\t0\t46844\t0\t0\n",
    "ncbi36:\n",
    "rs2691310: 1:46844 (forward strand) Jump to region in detail\n",
    "\n",
    "IMAGEN: all_subjects_all_snps_wave1.bim\n",
    "bim:\n",
    "1\trs4654470\t0\t4896086\tG\tA\n",
    "ncbi36:\n",
    "1:4896086 (forward strand) Jump to region in detail\n",
    "\n",
    "IMAGEN: all_subjects_all_snps_wave2.bim\n",
    "bim:\n",
    "1\trs945322\t0\t5606888\tC\tA\n",
    "ncbi36:\n",
    "1:5606888 (forward strand) Jump to region in detail\n",
    "\n",
    "##Get the genome position of a SNP in hg19:\n",
    "mysql --user=genome --host=genome-mysql.cse.ucsc.edu -A -D hg19 -e 'SELECT name,chromStart,chromEnd,alleles,alleleFreqs FROM snp138Common WHERE name = \"rs945322\"'\n",
    "\n",
    "##Get the genome position of a SNP in hg18:\n",
    "mysql --user=genome --host=genome-mysql.cse.ucsc.edu -A -D hg18 -e 'SELECT name,chrom,chromEnd FROM snp130 WHERE name = \"rs945322\"'\n",
    "\n",
    "All studies appear to be in hg18 with the exception of ADNI2 (ADNI_GO) and Osaka which are hg19"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
