{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Simple Genomic Analyses Using the ADNI dataset\n",
    "Roberto Toro, ValparaÃ­so 2016"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "We will use public data from the ADNI project (Alzheimer Disease Neuroimaging Initiative). We will obtain demographic, phenotypic and genetic variables from here:\n",
    "\n",
    "http://adni.loni.usc.edu/"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "Genome: chromosomes, genes, introns, exons\n",
    "Evolution: Allele frequency, linkage disequilibrium, population structure"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "@----------------------------------------------------------@\n",
      "|        PLINK!       |     v1.07      |   10/Aug/2009     |\n",
      "|----------------------------------------------------------|\n",
      "|  (C) 2009 Shaun Purcell, GNU General Public License, v2  |\n",
      "|----------------------------------------------------------|\n",
      "|  For documentation, citation & bug-report instructions:  |\n",
      "|        http://pngu.mgh.harvard.edu/purcell/plink/        |\n",
      "@----------------------------------------------------------@\n",
      "\n",
      "Web-based version check ( --noweb to skip )\n",
      "Recent cached web-check found... OK, v1.07 is current\n",
      "\n",
      "+++ PLINK 1.9 is now available! See above website for details +++ \n",
      "\n",
      "Writing this text to log file [ Simple_Genomics/out.log ]\n",
      "Analysis started: Sun Dec 11 08:23:21 2016\n",
      "\n",
      "Options in effect:\n",
      "\t--bfile Simple_Genomics/adni100k\n",
      "\t--out Simple_Genomics/out\n",
      "\n",
      "Reading map (extended format) from [ Simple_Genomics/adni100k.bim ] \n",
      "100257 markers to be included from [ Simple_Genomics/adni100k.bim ]\n",
      "Reading pedigree information from [ Simple_Genomics/adni100k.fam ] \n",
      "757 individuals read from [ Simple_Genomics/adni100k.fam ] \n",
      "0 individuals with nonmissing phenotypes\n",
      "Assuming a disease phenotype (1=unaff, 2=aff, 0=miss)\n",
      "Missing phenotype value is also -9\n",
      "0 cases, 0 controls and 757 missing\n",
      "449 males, 308 females, and 0 of unspecified sex\n",
      "Reading genotype bitfile from [ Simple_Genomics/adni100k.bed ] \n",
      "Detected that binary PED file is v1.00 SNP-major mode\n",
      "Before frequency and genotyping pruning, there are 100257 SNPs\n",
      "757 founders and 0 non-founders found\n",
      "1020 heterozygous haploid genotypes; set to missing\n",
      "Writing list of heterozygous haploid genotypes to [ Simple_Genomics/out.hh ]\n",
      "4545 SNPs with no founder genotypes observed\n",
      "Warning, MAF set to 0 for these SNPs (see --nonfounders)\n",
      "Writing list of these SNPs to [ Simple_Genomics/out.nof ]\n",
      "Total genotyping rate in remaining individuals is 0.949356\n",
      "0 SNPs failed missingness test ( GENO > 1 )\n",
      "0 SNPs failed frequency test ( MAF < 0 )\n",
      "After frequency and genotyping pruning, there are 100257 SNPs\n",
      "After filtering, 0 cases, 0 controls and 757 missing\n",
      "After filtering, 449 males, 308 females, and 0 of unspecified sex\n",
      "\n",
      "Analysis finished: Sun Dec 11 08:23:25 2016\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "geno=\"Simple_Genomics/adni100k\"\n",
    "plink=\"Simple_Genomics/plink-1.07-mac-intel/plink\"\n",
    "$plink --bfile $geno --out \"Simple_Genomics/out\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "@----------------------------------------------------------@\n",
      "|        PLINK!       |     v1.07      |   10/Aug/2009     |\n",
      "|----------------------------------------------------------|\n",
      "|  (C) 2009 Shaun Purcell, GNU General Public License, v2  |\n",
      "|----------------------------------------------------------|\n",
      "|  For documentation, citation & bug-report instructions:  |\n",
      "|        http://pngu.mgh.harvard.edu/purcell/plink/        |\n",
      "@----------------------------------------------------------@\n",
      "\n",
      "Web-based version check ( --noweb to skip )\n",
      "Recent cached web-check found... OK, v1.07 is current\n",
      "\n",
      "+++ PLINK 1.9 is now available! See above website for details +++ \n",
      "\n",
      "Writing this text to log file [ Simple_Genomics/adni100kFiltered.log ]\n",
      "Analysis started: Sun Dec 11 12:21:59 2016\n",
      "\n",
      "Options in effect:\n",
      "\t--bfile Simple_Genomics/adni100k\n",
      "\t--out Simple_Genomics/adni100kFiltered\n",
      "\t--maf 0.05\n",
      "\t--geno 0.01\n",
      "\t--hwe 1e-6\n",
      "\t--mind 0.1\n",
      "\t--make-bed\n",
      "\n",
      "Reading map (extended format) from [ Simple_Genomics/adni100k.bim ] \n",
      "100257 markers to be included from [ Simple_Genomics/adni100k.bim ]\n",
      "Reading pedigree information from [ Simple_Genomics/adni100k.fam ] \n",
      "757 individuals read from [ Simple_Genomics/adni100k.fam ] \n",
      "0 individuals with nonmissing phenotypes\n",
      "Assuming a disease phenotype (1=unaff, 2=aff, 0=miss)\n",
      "Missing phenotype value is also -9\n",
      "0 cases, 0 controls and 757 missing\n",
      "449 males, 308 females, and 0 of unspecified sex\n",
      "Reading genotype bitfile from [ Simple_Genomics/adni100k.bed ] \n",
      "Detected that binary PED file is v1.00 SNP-major mode\n",
      "Before frequency and genotyping pruning, there are 100257 SNPs\n",
      "757 founders and 0 non-founders found\n",
      "Writing list of removed individuals to [ Simple_Genomics/adni100kFiltered.irem ]\n",
      "1 of 757 individuals removed for low genotyping ( MIND > 0.1 )\n",
      "963 heterozygous haploid genotypes; set to missing\n",
      "Writing list of heterozygous haploid genotypes to [ Simple_Genomics/adni100kFiltered.hh ]\n",
      "433 markers to be excluded based on HWE test ( p <= 1e-06 )\n",
      "\t0 markers failed HWE test in cases\n",
      "\t0 markers failed HWE test in controls\n",
      "Total genotyping rate in remaining individuals is 0.949437\n",
      "15461 SNPs failed missingness test ( GENO > 0.01 )\n",
      "13808 SNPs failed frequency test ( MAF < 0.05 )\n",
      "After frequency and genotyping pruning, there are 76551 SNPs\n",
      "After filtering, 0 cases, 0 controls and 756 missing\n",
      "After filtering, 448 males, 308 females, and 0 of unspecified sex\n",
      "Writing pedigree information to [ Simple_Genomics/adni100kFiltered.fam ] \n",
      "Writing map (extended format) information to [ Simple_Genomics/adni100kFiltered.bim ] \n",
      "Writing genotype bitfile to [ Simple_Genomics/adni100kFiltered.bed ] \n",
      "Using (default) SNP-major mode\n",
      "\n",
      "Analysis finished: Sun Dec 11 12:22:07 2016\n",
      "\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "bash: line 5: fg: no job control\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "geno=\"Simple_Genomics/adni100k\"\n",
    "plink=\"Simple_Genomics/plink-1.07-mac-intel/plink\"\n",
    "\n",
    "# Filter SNPs with standard criteria for quality\n",
    "%%bash\n",
    "geno=\"Simple_Genomics/adni100k\"\n",
    "plink=\"Simple_Genomics/plink-1.07-mac-intel/plink\"\n",
    "$plink --bfile $geno --out \"Simple_Genomics/adni100kFiltered\" --maf 0.05 --geno 0.01 --hwe 1e-6 --mind 0.1 --make-bed"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "@----------------------------------------------------------@\n",
      "|        PLINK!       |     v1.07      |   10/Aug/2009     |\n",
      "|----------------------------------------------------------|\n",
      "|  (C) 2009 Shaun Purcell, GNU General Public License, v2  |\n",
      "|----------------------------------------------------------|\n",
      "|  For documentation, citation & bug-report instructions:  |\n",
      "|        http://pngu.mgh.harvard.edu/purcell/plink/        |\n",
      "@----------------------------------------------------------@\n",
      "\n",
      "Web-based version check ( --noweb to skip )\n",
      "Recent cached web-check found... OK, v1.07 is current\n",
      "\n",
      "+++ PLINK 1.9 is now available! See above website for details +++ \n",
      "\n",
      "Writing this text to log file [ Simple_Genomics/out.log ]\n",
      "Analysis started: Sun Dec 11 12:22:20 2016\n",
      "\n",
      "Options in effect:\n",
      "\t--bfile Simple_Genomics/adni100kFiltered\n",
      "\t--out Simple_Genomics/out\n",
      "\t--freq\n",
      "\n",
      "Reading map (extended format) from [ Simple_Genomics/adni100kFiltered.bim ] \n",
      "76551 markers to be included from [ Simple_Genomics/adni100kFiltered.bim ]\n",
      "Reading pedigree information from [ Simple_Genomics/adni100kFiltered.fam ] \n",
      "756 individuals read from [ Simple_Genomics/adni100kFiltered.fam ] \n",
      "0 individuals with nonmissing phenotypes\n",
      "Assuming a disease phenotype (1=unaff, 2=aff, 0=miss)\n",
      "Missing phenotype value is also -9\n",
      "0 cases, 0 controls and 756 missing\n",
      "448 males, 308 females, and 0 of unspecified sex\n",
      "Reading genotype bitfile from [ Simple_Genomics/adni100kFiltered.bed ] \n",
      "Detected that binary PED file is v1.00 SNP-major mode\n",
      "Before frequency and genotyping pruning, there are 76551 SNPs\n",
      "756 founders and 0 non-founders found\n",
      "53 heterozygous haploid genotypes; set to missing\n",
      "Writing list of heterozygous haploid genotypes to [ Simple_Genomics/out.hh ]\n",
      "Writing allele frequencies (founders-only) to [ Simple_Genomics/out.frq ] \n",
      "\n",
      "Analysis finished: Sun Dec 11 12:22:23 2016\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "geno=\"Simple_Genomics/adni100kFiltered\"\n",
    "plink=\"Simple_Genomics/plink-1.07-mac-intel/plink\"\n",
    "$plink --bfile $geno --out \"Simple_Genomics/out\" --freq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Populating the interactive namespace from numpy and matplotlib\n"
     ]
    }
   ],
   "source": [
    "%pylab inline"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import sys\n",
    "import urllib2 as url\n",
    "import re\n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "import statsmodels.api as sm\n",
    "import statsmodels.formula.api as smf\n",
    "import statsmodels.stats.power as smp\n",
    "import matplotlib.pyplot as plt\n",
    "from StringIO import StringIO\n",
    "import json\n",
    "\n",
    "# Install seaborn typing `conda install seaborn`\n",
    "import seaborn as sns\n",
    "\n",
    "# printf is an old friend from C\n",
    "def printf(format, *args):\n",
    "    sys.stdout.write(format % args)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      " CHR          SNP   A1   A2          MAF  NCHROBS\n",
      "   1   rs11807848    C    T       0.3922     1512\n",
      "   1    rs9442373    C    A        0.439     1508\n",
      "   1   rs12145826    A    G      0.06897     1508\n",
      "   1    rs3813199    T    C        0.083     1506\n",
      "   1   rs12410087    C    A        0.109     1504\n",
      "   1    rs2649588    G    A      0.07672     1512\n",
      "   1    rs3128342    T    G       0.4848     1510\n",
      "   1    rs2296716    A    G       0.1197     1512\n",
      "   1   rs35672141    A    G        0.463     1512\n",
      "   1    rs7525092    T    C       0.2513     1512\n",
      "   1    rs2474460    A    G       0.4544     1512\n",
      "   1    rs2803329    A    G       0.2082     1508\n",
      "   1    rs2803291    A    G        0.203     1512\n",
      "   1   rs13303016    C    T       0.1573     1500\n",
      "   1    rs6605074    T    C        0.207     1512\n",
      "   1     rs884080    A    G       0.4987     1512\n",
      "   1    rs7553178    G    A       0.3073     1510\n",
      "   1    rs2055204    G    A       0.4616     1512\n",
      "   1    rs4648831    T    C       0.1951     1512\n"
     ]
    }
   ],
   "source": [
    "# Minor-allele frequency (MAF)\n",
    "f1 = open('Simple_Genomics/out.frq', 'r')\n",
    "for i in range(0,20):\n",
    "    printf(\"%s\",f1.readline())\n",
    "f1.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<matplotlib.axes._subplots.AxesSubplot at 0x11f82c850>"
      ]
     },
     "execution_count": 30,
     "metadata": {},
     "output_type": "execute_result"
    },
    {
     "data": {
      "image/png": "iVBORw0KGgoAAAANSUhEUgAAAZkAAAECCAYAAAAsBKpOAAAABHNCSVQICAgIfAhkiAAAAAlwSFlz\nAAALEgAACxIB0t1+/AAAFoJJREFUeJzt3XuYXXV97/H3MEmg6CRWDXJRQbD94tEejuANGyFcohKk\nYPsoVNsiFhDEIH1Ki0Y5Cs8TtVJ5Cr2gRRSQ0gpYrZRLUFGZgFrg4KFR/Bq59SmHayA3EkIuc/5Y\nK2EzTmZ2kvnttWfP+/VPZv/2b+313St78slav9/67b6hoSEkSSphh6YLkCT1LkNGklSMISNJKsaQ\nkSQVY8hIkooxZCRJxUwpvYOIeDPwucw8JCL2AS4FNgKLM/O0us9JwMnAOmBBZl4XETsBVwC7ACuA\n4zNzaUS8Bfibuu93MvPc0u9BkrRtip7JRMRfABcDO9ZN5wPzM/NgYIeIODoiXgbMAw4E3gl8NiKm\nAqcCd2fmQcDXgLPr17gIOC4z3wa8OSL2K/keJEnbrvTlsl8B7255fEBmDtY/3wDMAd4ELMrM9Zm5\nAlgC7AfMAm5s6XtYRAwA0zLzgbp9IXB42bcgSdpWRUMmM78JrG9p6mv5eSUwHRgAlre0rwJmDGtf\n2dK2YthrzBjfqiVJ46XTA/8bW34eAJZRhcb0Ye1P1e0Dw/quHKHvslLFSpK2T/GB/2H+T0QclJm3\nAEcANwO3AwsiYhrwG8C+wGLgNmAucEf952BmroyItRHxKuAB4B3Ap8fa6dDQ0FBfX99Y3SRJz7fd\n/3B2OmTOBC6uB/bvAa7JzKGIuBBYRPWG5mfmsxFxEXBZRAwCa4H31a9xCnAl1VnYTZl5+1g77evr\n4/HHVxZ4O9tn5syBrqvLmtpjTe3rxrqsqT0zZw6M3WkMfZNkFeahbvvLg+79UFnT2Kypfd1YlzW1\nZ+bMge0+k/FmTElSMYaMJKkYQ0aSVIwhI0kqxpCRJBXT6SnMkjShbdiwgQceuG9cX3OvvfYe19fr\nJoaMJG2FBx64j4+e9212nrHLuLze6uWPccFf/B677rr/qP3uuutOTj/9FD796c9w2GFzNrcff/xx\nRLyG+fM/xRNPPM5xx72bT37yHGbPPgyARx55eHOfoaEh+vr62H//N/CBD5w4LvWPxZCRpK2084xd\neOFv7tHx/e65515873s3bQ6Z++77Fc8888zm56+//lre854/5F//9erNIQPwqlftw4UXfrHj9YJj\nMpI0Yeyzz2/xyCMPs3r10wAsXHgDb3/7EZufv+mmGzjuuPezbt067r//uUt6Td50b8hI0gQye/ah\n/PCH3wfgnnt+xu/8TvWVWnfc8R/svfermTHjRRx55FF84xtXbd7mgQfu4/TTT2HevA9x+umn8MQT\nT3SsXi+XSdIE0dfXx5w57+S88z7Lbrvtzn77vX7zWcq1136Thx9+iDPPPJ1169Zx771L+PCH5wHN\nXi4zZCRpAtltt9155pk1XHPN1znllI/w0EP/zbJly7j//nu5+upvb+73+c8v4Prrr2XWrIMbvVxm\nyEjSVlq9/LFGX+uww+awcOENvPzlr+Chh/6bu+++i6OOevfz+hx11DEsWPBpZs06mCa/6sRVmBvU\npauuWlMbrKl93VjX9tRU6j6ZXXd9UTcepwn3fTKSNKH19/ezzz6/1XQZE4azyyRJxXgm0yEjnWI/\n9dQLefLJVR2tY6+99qa/v7+j+5Q0eRkyHTLeS1Fsi03LV3iqL6lTDJkOamopCklqimMykqRiDBlJ\nUjGT4nLZ92+5jQcffLjRGpY+8Xij+5ekJkyKkLn4qltYtkOzXwq08sHFTHnJ/2i0BknqtEkRMv1T\nptLfv2OjNezQP7XR/UtSEyZFyKgytHEj//VfD47ap1P37ni/jjQ5GDKTyJqVj/OFrz/BzjOaHZ/y\nfh1p8jBkJhnv1ZHUSU5hliQVY8hIkooxZCRJxRgykqRiDBlJUjGGjCSpGENGklSMISNJKsaQkSQV\nY8hIkorp+LIyETEFuAzYC1gPnARsAC4FNgKLM/O0uu9JwMnAOmBBZl4XETsBVwC7ACuA4zNzaYff\nhiSpDU2sXTYX6M/M342Iw4HPAFOB+Zk5GBEXRcTRwI+BecD+wM7Aooi4CTgVuDszz42IY4GzgTMa\neB/aRu2sBt2q1MrQrgQtlddEyPwSmBIRfcAMqrOUN2fmYP38DcDbqc5qFmXmemBFRCwB9gNmAX/V\n0vfsThav7dcNq0G7ErTUGU2EzCrgVcAvgJcARwFva3l+JTAdGACWD9tuxrD2TX01wbgatDQ5NBEy\nfwbcmJmfiIg9gB8A01qeHwCWUY23TB/W/lTdPjCsb9fr79+BoaaL0PO8+MUvZObMgbE7bsH2bFtK\nN9YE3VmXNXVGEyHzJNUlMqgCYgpwV0QcnJk/BI4AbgZuBxZExDTgN4B9gcXAbVTjOnfUfw4yAWzY\nsNGpfF3mySdX8fjjK7dp25kzB7Z521K6sSbozrqsqT3jEXpNhMzfAF+JiFuoBvw/BtwJfDkipgL3\nANdk5lBEXAgsAvqoJgY8GxEXAZdFxCCwFnhfA+9BktSGjodMZj4NHDvCU7NH6HsJcMmwtjXAe4sU\nJ0kaV17BkSQVY8hIkooxZCRJxRgykqRimphdJjVua5e2GW48l7pxeRv1MkNGk1I3LG0DLm+j3mfI\naNJyaRupPMdkJEnFGDKSpGIMGUlSMYaMJKkYQ0aSVIwhI0kqxinMUoO296bQVttzg6g3hKoUQ0Zq\nUDfcFOoNoSrJkJEa5k2h6mWOyUiSijFkJEnFGDKSpGIck5EmufGc4TaSrZn15iy33mPISJNcN8xw\nA2e59SpDRpIz3FSMYzKSpGI8k5HUFUqPDbUabZzIcaHxZchI6grdMDb09LJHOPO41/PKV+7Z8X0P\nD75eCTtDRlLXaHpsaPXyR/nC1/+vkyDGkSEjSS2aDjro7KXD0cycuf92v4YhI0ldphsuHa5e/hg/\n+YYhI0k9qRvOqMaDU5glScUYMpKkYgwZSVIxhowkqRhDRpJUjCEjSSrGkJEkFdPIfTIR8THg94Cp\nwD8AtwCXAhuBxZl5Wt3vJOBkYB2wIDOvi4idgCuAXYAVwPGZubTjb0KSNKaOn8lExMHAgZn5VmA2\n8ErgfGB+Zh4M7BARR0fEy4B5wIHAO4HPRsRU4FTg7sw8CPgacHan34MkqT1NXC57B7A4Ir4FfBv4\nd2D/zBysn78BmAO8CViUmeszcwWwBNgPmAXc2NL38E4WL0lqX1uXyyLieuCrwLcyc9127vOlVGcv\n7wL2pgqa1rBbCUwHBoDlLe2rgBnD2jf1lSR1oXbPZD5HdclqSUT8fUS8cTv2uRRYWJ+h/BJ4hio8\nNhkAllGNt0wf1v5U3T4wrK8kqQu1FTKZeUtm/inwGuDHwDciYnFEnBERO27lPhdRBRYRsTvwAuB7\n9VgNwBHAIHA7MCsipkXEDGBfYDFwGzC37ju37tv1+vudyCdp8mn7X76ImA38HfAZqjGRjwK7Ul3u\naltmXgfcFRH/Afwb1UD+nwPnRMStVDPOrsnMR4ELqULpu1QTA54FLgJeFxGDwInAOVuz/6Zs2LCx\n6RIkqePaHZN5ELiPalzmI5m5pm7/AdUZx1bJzI+N0Dx7hH6XAJcMa1sDvHdr9ylJ6rx2z2QOBY7N\nzMsBIuLVAJm5ITO3/1ttJEk9qd2QOZLnpg3vAlwbESeXKUmS1CvaDZmTgbcBZOaDwAFUN0pKkrRF\n7YbMVGBty+NngaHxL0eS1EvaXbvsW8DNEXFV/fj32cpZZZKkyafd+2TOoppOHFR36V+YmZ8sWZgk\naeLbmjsE7wGuojqreTIiDipTkiSpV7R7n8zfA0cB97Y0D1FNbZYkaUTtjsm8HYhNN2FKktSOdi+X\n3Qf0lSxEktR72j2TeRL4eUTcRrVqMgCZ+cEiVUmSekK7IXMjz93xL0lSW9oKmcy8LCL2Al4LLARe\nkZn3lyxMkjTxtTUmExHHAtcCFwAvBn4UEX9UsjBJ0sTX7sD/WcBbgZWZ+RjweuDjxaqSJPWEdkNm\nQ2au3PQgMx8G/BYuSdKo2h34/1lEfASYGhH/C/gw8NNyZUmSekG7ZzKnAXsAa4CvACuogkaSpC1q\nd3bZ01RjMI7DSJLa1u7aZRv59e+PeTgzXz7+JUmSekW7ZzKbL6tFxFTgGODAUkVJknrD1iz1D0Bm\nrsvMq3EFZknSGNq9XPYnLQ/7qO78f7ZIRZKkntHuFOZDWn4eAp4Ajh3/ciRJvaTdMZkTShciSeo9\n7V4uu59fn10G1aWzoczce1yrkiT1hHYvl10JrAUuBtYB7wfeCHyiUF2SpB7Qbsi8IzPf0PL4goi4\nMzMfLFGUJKk3tDuFuS8iDt/0ICLeRbW0jCRJW9TumczJwOURsSvV2MwvgOOLVSVJ6gntzi67E3ht\nRLwUeCYzV5UtS5LUC9r9Zsw9I+I7wI+AF0bEzfXXMUuStEXtjsl8CTgPWAU8CvwzcHmpoiRJvaHd\nkHlpZt4EkJlDmXkxML1cWZKkXtBuyKyJiJdT35AZEbOo7puRJGmL2p1d9mfAvwP7RMRPgRcD7ylW\nlSSpJ7QbMi+jusP/t4F+4BeZ6SrMkqRRtRsyn8/M64CfjdeOI2IX4A7gcGADcCmwEVicmafVfU6i\nukdnHbAgM6+LiJ2AK4BdqG4IPT4zl45XXZKk8dNuyNwbEV8BfgKs2dSYmds0wywipgBfBFbXTecD\n8zNzMCIuioijgR8D84D9gZ2BRRFxE3AqcHdmnhsRxwJnA2dsSx2SpLJGHfiPiD3qH5dSrbj8Fqrv\nljkEmL0d+/1r4CLg/9Wvu39mDtbP3QDMAd4ELMrM9Zm5AlgC7AfMAm5s6Xs4kqSuNNaZzLVUAXBC\nRPx5Zn5he3cYER8AHsvM70TE/Lq5NexWUk2PHgCWt7SvAmYMa9/UV5LUhcaawtzX8vP7x2mfJwBz\nIuL7VGcmlwMzW54fAJZRjbdMH9b+VN0+MKyvJKkLjRUyrV9U1rfFXlshMw/OzEMy8xDgp8AfAzdE\nxEF1lyOAQeB2YFZETIuIGcC+wGLgNmBu3Xdu3bfr9fe3e0uSJPWOrfmXb6RvxhwvZwLnRsStwFTg\nmsx8FLgQWAR8l2piwLNUYzmvi4hB4ETgnIJ1jZsNGzY2XYIkddxYYzKvjYj76p/3aPl5XL52OTMP\nbXk4e4TnLwEuGda2Bnjv9uxXktQZY4XMb3ekCklSTxo1ZPx6ZUnS9nA0WpJUjCEjSSrGkJEkFWPI\nSJKKMWQkScUYMpKkYgwZSVIxhowkqRhDRpJUjCEjSSrGkJEkFWPISJKKMWQkScUYMpKkYgwZSVIx\nhowkqRhDRpJUjCEjSSrGkJEkFWPISJKKMWQkScUYMpKkYgwZSVIxhowkqRhDRpJUjCEjSSrGkJEk\nFWPISJKKMWQkScUYMpKkYgwZSVIxhowkqRhDRpJUjCEjSSrGkJEkFTOl0zuMiCnAV4C9gGnAAuDn\nwKXARmBxZp5W9z0JOBlYByzIzOsiYifgCmAXYAVwfGYu7fDbkCS1oYkzmT8CnsjMg4B3An8HnA/M\nz8yDgR0i4uiIeBkwDziw7vfZiJgKnArcXW//NeDsBt6DJKkNTYTMVTwXDP3AemD/zBys224A5gBv\nAhZl5vrMXAEsAfYDZgE3tvQ9vFOFS5K2Tscvl2XmaoCIGACuBj4B/HVLl5XAdGAAWN7SvgqYMax9\nU19JUhdqZOA/Il4B3Axclpn/QjUWs8kAsIxqvGX6sPan6vaBYX27Xn+/cywkTT4d/5evHmtZCPxl\nZl5WN98VEQfVPx8BDAK3A7MiYlpEzAD2BRYDtwFz675z675db8OGjWN3kqQe0/HLZcDHgRcBZ0fE\n/waGgI8Cf1sP7N8DXJOZQxFxIbAI6KOaGPBsRFwEXBYRg8Ba4H0NvAdJUhuaGJM5AzhjhKdmj9D3\nEuCSYW1rgPcWKU6SNK4cKJAkFWPISJKKMWQkScUYMpKkYgwZSVIxhowkqRhDRpJUjCEjSSrGkJEk\nFWPISJKKMWQkScUYMpKkYgwZSVIxhowkqRhDRpJUjCEjSSrGkJEkFWPISJKKMWQkScUYMpKkYgwZ\nSVIxhowkqRhDRpJUjCEjSSrGkJEkFWPISJKKMWQkScUYMpKkYgwZSVIxhowkqRhDRpJUjCEjSSrG\nkJEkFWPISJKKMWQkScUYMpKkYqY0XcC2iIg+4B+A/YBngBMz875mq5IkDTdRz2SOAXbMzLcCHwfO\nb7geSdIIJmrIzAJuBMjMnwBvaLYcSdJIJmrITAeWtzxeHxET9b1IUs+akGMywApgoOXxDpm5cUud\nNzz9OBvXrSlf1SjWP/0o6/tmNFrDmpVPAn2N1tAtdXRDDd1SRzfU0C11dEMN3VLH6uWPjcvrTNSQ\nuRV4F3BNRLwF+M/ROl958eea/9RI0iQ0UUPmm8CciLi1fnxCk8VIkkbWNzQ01HQNkqQe5WC5JKkY\nQ0aSVIwhI0kqxpCRJBUzUWeXbTbWOmYRcRRwNrAO+Gpmfrluv5Pnbui8PzP/tFM11X12Bm4CPpiZ\nvyy9Htu21FS3NXacIuIPgY9S/d39Z2Z+uBPr1m1LXXV7k8fqD4CzgI3AlZl5YdOfqZFqqtsb/d2r\n+30JWJqZ85s+TiPVVD8udpzaqSsizgBOBDbdMPMh4FftvJdWEz5kaFnHLCLeTLWO2TEAETGlfnwA\nsAa4NSL+jepmTjLz0E7XVNd1APBFYI92t2mipojYEZo5ThGxE3Au8LrMXBsRV0bEu4Cpo72PBuv6\nDjR2rHYAPkP1OV8N/DwirgAO3tI2Ddb0NDT3u1fX9iHgdcAP292m0zV14HevnboOAP44M+9qqfPd\nY72X4Xrhctlo65i9BliSmSsycx2wCDiIKoVfEBELI+K79cHqVE0A06j+Yn6xFds0UVOTx2kt8NbM\nXFs/nkL1P6dOrFu3LXU1dqzq1S5ek5mrgJdS/V4/O8b7aKqmRn/3IuJA4I3Al9rdpqGaSh+nMeui\nCpmPR8RgRJzV5ja/phdCZrR1zIY/txKYQfW/qfMy8x3AqcA/jfPaZ6OurZaZP8rMh3j+uhGl12Pb\nlppW09BxysyhzHwcICLmAS/IzO+O9T4arKuxY1XXtbH+X+ZPgR/U9TT9mRpe09M0eJwiYlfgU8BH\n6JLfvVFqKn2cRq2r9s/AKcAhwKyIOLKNbX5NL4TMaOuYraA6KJsMAMuAJcA/AWTmEmApsFuHahrP\nbUrX9EsaPE4R0RcR5wGHAb/fzjYN1tXosar3+83M3B3YEfgTqn8MGv1MjVBTk8fpPcBLgOuBjwHv\ni4imj9OWaip9nMaqC+CCzHwyM9fX9b2ebThWvRAytwJzAUZYx+we4NUR8aKImAa8DfgR8EHgC/U2\nu1MdtIc7VNN4blO6pqaP0z9SXf89puXyVOnjtK11NXasImIgIn5Qf8ahOmPYUG9z5Cjvo9M1baTB\n45SZf5uZb6zHOT5HNRnhcuC2LW3TYE2lj9OodUXEdGBxROxcTxA4FLiD6lht1Wdqwi8r0zJD4n/W\nTSdQXUt8QWZ+uT7F+xTVqeglmfnFiJgKfBXYk+qDf1Zm/rhTNbX0uxk4JZ8/u2zzNptmeDVYU2PH\nCbgTuB0YrJ8bAi4Avj18m/E8TttR13XAZcAraeAzFREnUs0Eeha4G5hX92vsM7WFmqbQHb97xwMx\nbHZZ0797rTUV/d1rp66IeD/VLMpngO9l5jnbcqwmfMhIkrpXL1wukyR1KUNGklSMISNJKsaQkSQV\nY8hIkooxZCRJxRgykqRiDBlJUjH/H+jCKbSq+vDCAAAAAElFTkSuQmCC\n",
      "text/plain": [
       "<matplotlib.figure.Figure at 0x10c820f10>"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "# Histogram of MAF\n",
    "maf_df=pd.read_csv('Simple_Genomics/out.frq',sep='\\s+',header=0)\n",
    "maf_df.plot(y='MAF',kind='hist')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "chr1 5816\n",
      "chr2 6232\n",
      "chr3 5252\n",
      "chr4 4597\n",
      "chr5 4723\n",
      "chr6 5149\n",
      "chr7 4185\n",
      "chr8 4310\n",
      "chr9 3691\n",
      "chr10 3928\n",
      "chr11 3716\n",
      "chr12 3702\n",
      "chr13 2858\n",
      "chr14 2457\n",
      "chr15 2226\n",
      "chr16 2330\n",
      "chr17 1969\n",
      "chr18 2325\n",
      "chr19 1244\n",
      "chr20 1926\n",
      "chr21 1083\n",
      "chr22 1098\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "geno=\"Simple_Genomics/adni100kFiltered\"\n",
    "plink=\"Simple_Genomics/plink-1.07-mac-intel/plink\"\n",
    "\n",
    "# Number of SNPs per chromosome\n",
    "\n",
    "for ((i=1;i<23;i++)); do\n",
    "    echo -n \"chr${i} \"\n",
    "    $plink --bfile $geno --chr $i|grep \"After frequency and genotyping pruning, there are\"|awk '{print $8}'\n",
    "done"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n",
      "@----------------------------------------------------------@\n",
      "|        PLINK!       |     v1.07      |   10/Aug/2009     |\n",
      "|----------------------------------------------------------|\n",
      "|  (C) 2009 Shaun Purcell, GNU General Public License, v2  |\n",
      "|----------------------------------------------------------|\n",
      "|  For documentation, citation & bug-report instructions:  |\n",
      "|        http://pngu.mgh.harvard.edu/purcell/plink/        |\n",
      "@----------------------------------------------------------@\n",
      "\n",
      "Web-based version check ( --noweb to skip )\n",
      "Recent cached web-check found... OK, v1.07 is current\n",
      "\n",
      "+++ PLINK 1.9 is now available! See above website for details +++ \n",
      "\n",
      "Writing this text to log file [ Simple_Genomics/out.log ]\n",
      "Analysis started: Sun Dec 11 12:23:26 2016\n",
      "\n",
      "Options in effect:\n",
      "\t--bfile Simple_Genomics/adni100kFiltered\n",
      "\t--out Simple_Genomics/out\n",
      "\t--chr 22\n",
      "\t--r2\n",
      "\t--matrix\n",
      "\n",
      "Reading map (extended format) from [ Simple_Genomics/adni100kFiltered.bim ] \n",
      "76551 markers to be included from [ Simple_Genomics/adni100kFiltered.bim ]\n",
      "Scan region on chromosome 22 from [ rs9604967 ] to [ rs6010061 ]\n",
      "Reading pedigree information from [ Simple_Genomics/adni100kFiltered.fam ] \n",
      "756 individuals read from [ Simple_Genomics/adni100kFiltered.fam ] \n",
      "0 individuals with nonmissing phenotypes\n",
      "Assuming a disease phenotype (1=unaff, 2=aff, 0=miss)\n",
      "Missing phenotype value is also -9\n",
      "0 cases, 0 controls and 756 missing\n",
      "448 males, 308 females, and 0 of unspecified sex\n",
      "Reading genotype bitfile from [ Simple_Genomics/adni100kFiltered.bed ] \n",
      "Detected that binary PED file is v1.00 SNP-major mode\n",
      "Before frequency and genotyping pruning, there are 1098 SNPs\n",
      "756 founders and 0 non-founders found\n",
      "Total genotyping rate in remaining individuals is 0.998317\n",
      "0 SNPs failed missingness test ( GENO > 1 )\n",
      "0 SNPs failed frequency test ( MAF < 0 )\n",
      "After frequency and genotyping pruning, there are 1098 SNPs\n",
      "After filtering, 0 cases, 0 controls and 756 missing\n",
      "After filtering, 448 males, 308 females, and 0 of unspecified sex\n",
      "Writing LD statistics to [ Simple_Genomics/out.ld ] \n",
      "\n",
      "Analysis finished: Sun Dec 11 12:23:54 2016\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "geno=\"Simple_Genomics/adni100kFiltered\"\n",
    "plink=\"Simple_Genomics/plink-1.07-mac-intel/plink\"\n",
    "\n",
    "# Linkage disequilibrium\n",
    "$plink --bfile $geno --out \"Simple_Genomics/out\" --chr 22 --r2 --matrix"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "ld_df=pd.read_csv('Simple_Genomics/out.ld',sep='\\s+',header=None)\n",
    "plt.figure(figsize=(10, 10))\n",
    "plt.imshow(ld_df,cmap=plt.cm.jet)\n",
    "plt.colorbar()\n",
    "plt.show()\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Genome-wide association\n",
    "### Plink\n",
    "\n",
    "Genome-wide ssociation is used to identify loci involved with complex traits.\n",
    "\n",
    "We can test millions of genetic markers for association with a trait in a sample of unrelated subjects."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "echo $a"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "## Genomic complex-trait analysis\n",
    "### Genomic REML"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Linkage-disequilibrium score"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "anaconda-cloud": {},
  "kernelspec": {
   "display_name": "Python [Root]",
   "language": "python",
   "name": "Python [Root]"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
